### PR TITLE
Unblock the load event when the pdf has a password (bug 1801341)

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: web/app.js [hunk 1]
```diff
@@ -943,6 +943,11 @@ const PDFViewerApplication = {
     this.pdfLoadingTask = loadingTask;
 
     loadingTask.onPassword = (updateCallback, reason) => {
+      // The load event can't be triggered until the password is entered, so if
+      // the viewer is in an iframe and its visibility depends on the onload
+      // callback then the viewer can never shows (bug 1801341).
```
Comment (Snuffleupagus) [inline web/app.js:948]: Nit:
```suggestion
      // callback then the viewer never shows (bug 1801341).
```

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
Note that the reason for delaying the "load" event was so that automatic printing of PDF documents in `iframe`-elements still works; please see https://bugzilla.mozilla.org/show_bug.cgi?id=1618553
With these changes we basically regress that specifically for password-protected PDF documents, is that actually what we want to do here?

**Comment by calixteman** [comment]
> Note that the reason for delaying the "load" event was so that automatic printing of PDF documents in `iframe`-elements still works; please see https://bugzilla.mozilla.org/show_bug.cgi?id=1618553 With these changes we basically regress that specifically for password-protected PDF documents, is that actually what we want to do here?

I'm not super excited by the idea to break something...
Do you know if there is a way to know that an element with a specific id (here the dialog password) is visible within an iframe (I had a look at `IntersectionObserver` but I didn't to make it working within an iframe created with `createElement`) ?

**Comment by Snuffleupagus** [comment]
> Do you know if there is a way to know that an element with a specific id (here the dialog password) is visible within an iframe (I had a look at `IntersectionObserver` but I didn't to make it working within an iframe created with `createElement`) ?

No idea unfortunately :-(

---

However, if we implement this with the tweak suggested in https://github.com/mozilla/pdf.js/pull/15727#discussion_r1029797641 we should be limiting any problems considerably.
Given that you can't really trigger auto-printing on a *password-protected* PDF document embedded in an `iframe` using its "load"-event, since that'd always require the user to manually input a password, we should be pretty safe (at least I *think* so).

**Review by Snuffleupagus — APPROVED** [review]
r=me, with the nit from https://github.com/mozilla/pdf.js/pull/15727#discussion_r1029651886 taken care of; thank you!

**Pull request merged**